lxd/db/instances.go

```diff
@@ -872,6 +872,49 @@ func (c *ClusterTx) GetInstanceSnapshotsWithName(ctx context.Context, project st
 	return instances, nil
 }
 
+// GetInstancesWithConfig returns all available instances with the given config key and value.
+func (c *ClusterTx) GetInstancesWithConfig(key string, value string) ([]cluster.Instance, error) {
+	q := `
+SELECT instances.id, projects.name AS project, instances.name, nodes.name AS node, instances.type, instances.architecture, instances.ephemeral, instances.creation_date, instances.stateful, instances.last_use_date, coalesce(instances.description, ''), instances.expiry_date
+  FROM instances JOIN projects ON instances.project_id = projects.id JOIN nodes ON instances.node_id = nodes.id JOIN instances_config ON instances.id = instances_config.instance_id
+  WHERE instances_config.key = ? AND instances_config.value = ? ORDER BY projects.id, instances.name
+  `
+
+	stmt, err := c.tx.Prepare(q)
```

Comment:
We should never use Prepare when using a function-local SQL queries, as otherwise we are incurring the cost of preparing the statement as well as the cost of actually running the query on every function call.

Instead use `tx.QueryScan()` for easily running a single adhoc query that allows you to scan multiple rows without loading them all into memory first.